test/e2e_node/builder/build.go

```diff
@@ -47,6 +47,10 @@ func BuildGo() error {
 	}
 	targets := strings.Join(buildTargets, " ")
 	cmd := exec.Command("make", "-C", k8sRoot, fmt.Sprintf("WHAT=%s", targets))
+	if utils.IsBuildForArm64() {
+		klog.Infof("Building k8s binaries for Arm64: %s", targets)
+		cmd = exec.Command("make", "-C", k8sRoot, fmt.Sprintf("WHAT=%s", targets), "KUBE_BUILD_PLATFORMS=linux/arm64")
```

Comment:
Note: this assumes `C compiler "aarch64-linux-gnu-gcc" ` is available on the build machine. 

Can be installed with `sudo apt-get install gcc-aarch64-linux-gnu`

Reply:
Thanks for the pointer. Added this to kubekins Dockerfile: https://github.com/kubernetes/test-infra/pull/29197

Reply:
you already need to run docker right? you can use the dockerized make build here instead

Reply:
looks like no, the CI job does not use docker

Reply:
Thinking about this some more, it seems like it would be beneficial to switch to use dockerized make build here. The benefit is that  dockerized make build already uses the kubecross image which supports building for arm (and has the all the needed cross build dependencies already installed). 

I would suggest the following:

1. Let's add a flag to the node e2e runner which controls if we use the dockerized build (e.g. `--use-dockerized-build`).
2. If the flag is enabled, build the artifacts here with ` build/run.sh make WHAT=cmd/kubelet KUBE_BUILD_PLATFORMS=linux/arm64` (+ the other targets needed).
3. In the test job for ARM, we enable  `--use-dockerized-build` and also set `privileged: true` since it requires nested docker

This also means we don't need to modify the kubekins image.

Long term, ideally we can switch all the jobs over to use dockerized build, it would be ideal as we already have some issues with issues like glibc incompatibility, but it will take some time to update all the jobs.

Reply:
I think this should be a pre-req. We do not want to duplicate maintaining the build environment, and the kubekins image is already _massive_

Reply:
Done. Updated the current implementation with `--use-dockerized-build` flag and enable dockerized build instead of kubekins image.